### PR TITLE
Add MoCmaSampler

### PR DESCRIPTION
**Pull request by y0z**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

> [!TIP]
> Please follow the [Quick TODO list](https://github.com/optuna/optunahub-registry/tree/main?tab=readme-ov-file#quick-todo-list-towards-contribution) to smoothly merge your PR.

## Motivation & Description of the changes

Add `MoCmaSampler`, which implements [the CMA-ES variant for multi-objective optimization](https://doi.org/10.1162/evco.2007.15.1.1).

## TODO List towards PR Merge

Please remove this section if this PR is not an addition of a new package.
Otherwise, please check the following TODO list:

- [x] Copy `./template/` to create your package
- [x] Replace `<COPYRIGHT HOLDER>` in `LICENSE` of your package with your name
- [x] Fill out `README.md` in your package
- [x] Add import statements of your function or class names to be used in `__init__.py`
- [x] Apply the formatter based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)
- [x] Check whether your module works as intended based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@nabenabe0928 Could you review this PR?

File: package/samplers/mocma/mocma.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
+from collections.abc import Sequence
```
Comment (nabenabe0928) [inline package/samplers/mocma/mocma.py:1]: This sampler does not work with Python 3.9 without future annotations.

```suggestion
from __future__ import annotations

from collections.abc import Sequence
```

**Review by nabenabe0928 — COMMENTED** [review]
Almost LGTM!

File: package/samplers/mocma/mocma.py (hunk 1 above)
Comment (y0z) [reply package/samplers/mocma/mocma.py:3]: Thank you for your comment. I applied your suggestion.

**Review by nabenabe0928 — APPROVED** [review]
LGTM

**Pull request merged**